library/include/rocwmma/internal/transforms_impl.hpp

```diff
@@ -211,6 +226,284 @@ namespace rocwmma
         }
     };
 
+    template <>
+    struct AosToSoa<16, 16>
+    {
+        constexpr static uint32_t VW       = 16;
+        constexpr static uint32_t VecSize  = 16;
+        constexpr static uint32_t BlockDim = 16;
+
+        template <typename DataT>
+        ROCWMMA_DEVICE constexpr static inline auto exec(VecT<DataT, VecSize> const& v)
+        {
+            using PackUtil = PackUtil<DataT>;
+
+            // Step 1 : Unpack groups of 1
+            auto result = unpackLoHi1(v);
+
+            // Step 2 : Unpack groups of 2
+            result = unpackLoHi2(result);
+
+            // Step 3 : Unpack groups of 4
+            result = unpackLoHi4(result);
+
+            // Step 4 : Unpack groups of 8
+            result = unpackLoHi8(result);
+
+            return result;
+        }
+    };
+
+    template <>
+    struct AosToSoa<32, 16>
+    {
+        constexpr static uint32_t VW      = 16;
+        constexpr static uint32_t VecSize = 16;
+
+        template <typename DataT>
+        ROCWMMA_DEVICE constexpr static inline auto exec(VecT<DataT, VecSize> const& v)
+        {
+            using PackUtil = PackUtil<DataT>;
+
+            // Step 1 : Unpack groups of 2
+            auto result = unpackLoHi2(v);
+
+            // Step 2 : Unpack groups of 4
+            result = unpackLoHi4(result);
+
+            // Step 3 : Unpack groups of 8
+            result = unpackLoHi8(result);
+
+            // Step 4 : Unpack groups of 16
+            result = unpackLoHi16(result);
+
+            // Step 5 : Gather
+            auto lo = PackUtil::paddedPack(extractLo(result));
```

Comment:
Because the Gather32 operation is the same here, we can just pack the entire result and perform a single gather

Reply:
Addressed in latest commit